backend/tests/clubs/test_ticketing.py

```diff
@@ -947,6 +947,72 @@ def test_get_cart_replacement_required_sold_out(self):
         to_add = set(map(lambda t: str(t.id), tickets_to_add))
         self.assertEqual(len(in_cart & to_add), 0, in_cart | to_add)
 
+    def test_place_hold_on_tickets(self):
+        from clubs.views import TicketViewSet
+
+        self.client.login(username=self.user1.username, password="test")
+
+        # Add a few tickets
+        cart, _ = Cart.objects.get_or_create(owner=self.user1)
+        tickets_to_add = self.tickets1[:3]
+        for ticket in tickets_to_add:
+            cart.tickets.add(ticket)
+        cart.save()
+
+        TicketViewSet._place_hold_on_tickets(self.user1, cart.tickets)
+        holding_expiration = timezone.now() + timezone.timedelta(minutes=10)
+
+        for ticket in cart.tickets.all():
+            self.assertIsNone(ticket.owner)
+            self.assertEqual(self.user1, ticket.holder)
+            self.assertAlmostEqual(
+                holding_expiration,
+                ticket.holding_expiration,
+                delta=timedelta(seconds=10),
+            )
+
+        # Move Django's internal clock 10 minutes forward
+        with freezegun.freeze_time(holding_expiration):
+            Ticket.objects.update_holds()
+            for ticket in cart.tickets.all():
+                self.assertIsNone(ticket.owner)
+                self.assertIsNone(ticket.holder)
+                # TODO: Ask whether to reset holding expiration to None
+
+    def test_give_tickets(self):
+        from clubs.views import TicketViewSet
+
+        self.client.login(username=self.user1.username, password="test")
+        # Add a few tickets
+        cart, _ = Cart.objects.get_or_create(owner=self.user1)
+        tickets_to_add = self.tickets1[:3]
+        for ticket in tickets_to_add:
+            cart.tickets.add(ticket)
+        cart.save()
+
+        order_info = {
+            "amountDetails": {"totalAmount": TicketViewSet._calculate_cart_total(cart)},
+            "billTo": {
+                "firstName": self.user1.first_name,
+                "lastName": self.user1.last_name,
+                "phoneNumber": None,
+                "email": self.user1.email,
+            },
+        }
+
+        TicketViewSet._place_hold_on_tickets(self.user1, cart.tickets)
+        TicketViewSet._give_tickets(
+            self.user1, order_info, cart, reconciliation_id=None
+        )
```

Comment:
Maybe also check for the creation of transaction records? Let me know what you think.

Reply:
I remember it being tested.
lines 1159 and 1338

Reply:
Yup, but IMO it's still good practice to move those tests over. Those tests were written back when the "give tickets" functionality wasn't its own function. Now that it is, unit tests for `_give_tickets` should test all its intended functionality. It's not needed here, but something to keep in mind with future unit tests.

Reply:
I see, will apply changes